platforms/software/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -219,7 +222,9 @@ public void setupProjectLayout() {
             insecureProtocol.get(),
             projectDir,
             toolChainVersion);
-        initDescriptor.generate(settings);
+        BuildContentGenerationContext buildContentGenerationContext = new BuildContentGenerationContext(new VersionCatalogDependencyRegistry());
+        initDescriptor.generate(settings, buildContentGenerationContext);
+        VersionCatalogGenerator.create(settings.getTarget()).generate(buildContentGenerationContext);
```

Comment:
❌ This seems incorrect to me.  Shouldn't the existing sub-types of the `initDescriptor` be modified to understand how to generate VCs?  I don't think this code in the task should be changing at all.  

It seems undesirable to have a second generator type be created and run immediately after the bulk of the project code is generated just to output the VC.  We've already had discussions about how it's bad the init is not "atomic", in that if it fails in the middle of generation it leaves a partial project on disk instead of cleaning up after itself - this seems like it would make it harder to address that problem and increase the likelihood of it happening.

Reply:
It's not that simple. (you already knew that 😅) I can see about adding a wrapper for the `initDescriptor` so that the code here doesn't have to change while at the same time making sure the version catalog generation is the very last thing that happens (it uses the state that emerges during the generation of the projects).

Reply:
A wrapper seems like a better idea.

Reply:
FTR: done